web/src/engine/websites/YugenMangasPT.ts

```diff
@@ -1,44 +1,77 @@
+import { Chapter, DecoratableMangaScraper, Page, type MangaPlugin, Manga } from '../providers/MangaPlugin';
 import * as Common from './decorators/Common';
 import { FetchJSON } from '../platform/FetchProvider';
 
-type APIPages = {
-    chapter_images: string[]
+type APIMangas = {
+    series: APIManga[]
 }
 
-function ChapterExtractor(anchor: HTMLAnchorElement) {
-    return {
-        id: anchor.pathname,
-        title: anchor.querySelector('.chapter-title').textContent.trim()
-    };
+type APIManga = {
+    name: string,
+    slug: string
+}
+
+type APIChapters = {
+    chapters: APIChapter[]
+}
+
+type APIChapter = {
+    name: string,
+    slug: string
+}
+
+type APIPages = {
+    chapter_images: string[]
 }
 
-@Common.MangaCSS(/^{origin}\/series\/[^/]+\/$/, 'div.sinopse div.title-name h1')
-@Common.MangasMultiPageCSS('/series?page={page}', 'div.gallery a.mangas-gallery')
-@Common.ChaptersSinglePageCSS('div#listadecapitulos div.chapter a', ChapterExtractor)
 @Common.ImageAjax()
 export default class extends DecoratableMangaScraper {
-    private readonly mediaUrl = 'https://media.yugenmangas.org/';
+    private readonly apiUrl = 'https://api.yugenmangas.net.br';
 
     public constructor() {
-        super('yugenmangas-pt', 'YugenMangas (PT)', 'https://yugenmangas.org', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Portuguese);
+        super('yugenmangas-pt', 'YugenMangas (PT)', 'https://yugenmangas.net.br', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Portuguese, Tags.Source.Scanlator);
     }
 
     public override get Icon() {
         return icon;
     }
 
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/series/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const slug = url.split('/').pop();
+        const data = await FetchJSON<APIManga>(new Request(new URL(`/api/serie_details/${slug}`, this.apiUrl), {
+            method: 'POST',
+        }));
+        return new Manga(this, provider, data.slug, data.name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const { series } = await FetchJSON<APIMangas>(new Request(new URL('/api/all_series/', this.apiUrl)));
+        return series.map(manga => new Manga(this, provider, manga.slug, manga.name.trim()));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const { chapters } = await FetchJSON<APIChapters>(new Request(new URL('/api/get_chapters_by_serie/', this.apiUrl), {
+            method: 'POST',
+            body: JSON.stringify({
+                serie_slug: manga.Identifier
+            })
+        }));
+        return chapters
+            .sort((chapterA, chapterB) => { return chapterB.name.localeCompare(chapterA.name, undefined, { numeric: true, sensitivity: 'base' }); })
```

Comment:
Is sorting necessary? In legacy HakuNeko chapters are presented 'as is' and the UI does the sorting.

Reply:
Yes it is. Api order is a real mess, and its sorted on the website there is obviously some sorting underneath.
Legacy Hakuneko YugenMangas is the spanish one, we are dealing with the PT one. 

Yugenmanga two websites are a pain to follow.